hashes/src/error.rs

```diff
@@ -17,17 +17,45 @@
 
 use core::fmt;
 
+use internals::write_err;
+
+use crate::hex;
+
 /// Crate error type.
 #[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub enum Error {
     /// Tried to create a fixed-length hash from a slice with the wrong size (expected, got).
     InvalidLength(usize, usize),
+    /// A hexadecimal parsing error.
+    Hex(hex::Error),
 }
```

Comment:
> This is not great, `hex::Error` already contains `InvalidLength` variant. I see this is because you wanted to use it in `FromStr::Error`. IMO we should have `struct ParseError(hex::Error);` for that.

I just changed the `FromStr` impls to use `hex::Error` instead of `hashes::Error` - any reason not to do that?

> This error type is provided only because we have redundant `from_slice()` methods for thing that can be achieved with `from_inner(x.try_into())`. I think we should consider removing both the type and the helper function.

I played around with this, had to return `core::alloc::TryFromSliceError` - all was well until the derives on `pstb::Error` bit me. I'm going to leave this for now.

Reply:
> I'm confused, the `FromStr` impls do use `hex::Error` currently on master already.

Yes, I meant we should encapsulate it to not leak internal type.

But thinking about it, long term we will have `hex` crate with `Error` being just invalid char and odd length - no invalid length. And since the crate will be public we could then have:

```rust
enum ParseError {
    Hex(hex::Error),
    InvalidLength(InvalidLengthError),
}

#[non_exhaustive]
struct InvalidLengthError {
    required: usize,
    provided: usize
}
```